sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVInferSchema.scala

```diff
@@ -32,7 +34,10 @@ class CSVInferSchema(val options: CSVOptions) extends Serializable {
     options.zoneId,
     options.locale)
 
-  private val decimalParser = {
+  private val decimalParser = if (options.locale == Locale.US) {
```

Comment:
It looks slightly ugly that we handle special case of `Locale.US` for JSON inside of `ExprUtils.getDecimalParser` but for CSV outside of it. Maybe we unify the implementation, and  handle the special cases outside of generic implementation `ExprUtils.getDecimalParser`?

Reply:
Yes.. we gotta do that I think .. Let's do that later when we drop the replacement thing entirely. I realised that CSV inference path alone handles it without replacement.